### PR TITLE
Gene Set feature branch PR

### PR DESCRIPTION
**Pull request by pieterlukasse**
# What? Why?
Adding support for Gene Set Variation Analysis (GSVA) to cBioPortal.

This PR is a combination of multiple PRs, reviewed, approved and merged into this `geneset` feature branch. For details, please refer to the individual PRs:

https://github.com/cBioPortal/cbioportal/pull/2155: Gene set related tables
https://github.com/cBioPortal/cbioportal/pull/2158: Gene set APIs
https://github.com/cBioPortal/cbioportal/pull/2166: Gene set Importer
https://github.com/cBioPortal/cbioportal/pull/2164: Gene set frontend
https://github.com/cBioPortal/cbioportal/pull/2186: Gene set oncoprint tracks
https://github.com/cBioPortal/cbioportal/pull/2190: Gene set oncoprint track expansion
https://github.com/cBioPortal/cbioportal/pull/2390: Gene set importer improvements

### CODE REVIEW

File: web/src/main/java/org/cbioportal/web/GenesetDataController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+@PublicApi
+@RestController
+@Validated
+@Api(tags = "Gene Set Genetic Data", description = " ")
```
Comment (ersinciftci) [inline web/src/main/java/org/cbioportal/web/GenesetDataController.java]: We can change the API name to `Gene Set Scores` if it makes more sense.

**Comment by ersinciftci** [comment]
It seems the database that the heroku instance of this PR connecting doesn't have geneset related tables, I get `Table 'public_test.geneset' doesn't exist` when I try to test the APIs. Can you add them?

**Comment by pieterlukasse** [comment]
hi @ersinciftci , feel free to test on https://cbioportal.thehyve.net/cbioportal/

File: web/src/main/java/org/cbioportal/web/GenesetDataController.java (hunk 1 above)
Comment (pieterlukasse) [inline web/src/main/java/org/cbioportal/web/GenesetDataController.java]: makes sense. Changed

**Comment by pieterlukasse** [comment]
after discussing with @jjgao we will be merging the first part here: https://github.com/cBioPortal/cbioportal/pull/2470

**Comment by pieterlukasse** [comment]
First part has been merged via #2470. Second part now in a separate PR: #2476. Therefore closing this one.

**Pull request closed without merging**